kernel/aster-nix/src/process/posix_thread/futex.rs

```diff
@@ -32,22 +34,23 @@ pub fn futex_wait_bitset(
         "futex_wait_bitset addr: {:#x}, val: {}, timeout: {:?}, bitset: {:#x}",
         futex_addr, futex_val, timeout, bitset
     );
-    let futex_key = FutexKey::new(futex_addr);
-    let (_, futex_bucket_ref) = get_futex_bucket(futex_key);
+    let futex_key = FutexKey::new(futex_addr, bitset);
+    if futex_key.load_val() != futex_val {
+        return_errno_with_message!(Errno::EAGAIN, "futex value does not match");
+    }
+    let (waiter, futex_item) = FutexItem::build(futex_key);
```

Comment:
I agree that the verb `new` would sound too light-weight, but the verb `build` sounds too heavy-weight. I suggest choosing the verb `create` or `alloc`. And since this is the constructor of `FutexItem`, I suppose that returning `FutexItem` at the first field of the result tuple is better.

```rust
    let (futex_item, waiter) = FutexItem::create(futex_key);
```